File: backend/app/schemas/screening.py
"""筛选相关的Pydantic schemas"""
from pydantic import BaseModel, Field
from typing import List, Optional, Dict
from uuid import UUID


class ScreeningRequest(BaseModel):
    """筛选请求"""
    resume_id: UUID = Field(..., description="简历ID")
    job_id: UUID = Field(..., description="岗位ID")


class ScreeningResponse(BaseModel):
    """筛选结果"""
    id: UUID
    resume_id: UUID
    job_id: UUID
    match_score: int
    skill_score: int
    experience_score: int
    education_score: int
    screening_result: str  # PASS / REVIEW / REJECT
    matched_points: List[str]
    unmatched_points: List[str]
    suggestion: Optional[str] = None

    class Config:
        from_attributes = True


class MatchRequest(BaseModel):
    """匹配请求（不保存到数据库）"""
    candidate_name: str = Field(..., description="候选人姓名")
    phone: Optional[str] = Field(None, description="电话")
    email: Optional[str] = Field(None, description="邮箱")
    education: Optional[str] = Field(None, description="学历")
    work_years: Optional[int] = Field(None, description="工作年限")
    skills: List[str] = Field(default_factory=list, description="技能列表")
    job_id: UUID = Field(..., description="岗位ID")

File: frontend/src/types/screening.ts
/** 筛选相关类型定义 */

export interface MatchRequest {
  candidate_name: string;
  phone?: string;
  email?: string;
  education?: string;
  work_years?: number;
  skills: string[];
  job_id: string;
}

export interface MatchResult {
  candidate_name: string;
  job_name: string;
  screening_result: 'PASS' | 'REVIEW' | 'REJECT';
  agent_score: number;
  matched_points: string[];
  unmatched_points: string[];
  suggestion: string;
}

export interface ScreeningResult {
  id: string;
  resume_id: string;
  job_id: string;
  agent_score: number | null;
  screening_result: 'PASS' | 'REVIEW' | 'REJECT';
  matched_points: string[];
  unmatched_points: string[];
  suggestion?: string;
  created_at: string;
}

// 筛选结果显示名称映射
export const SCREENING_RESULT_LABELS: Record<MatchResult['screening_result'], string> = {
  PASS: '通过',
  REVIEW: '待定',
  REJECT: '拒绝'
};

// 筛选结果颜色映射
export const SCREENING_RESULT_COLORS: Record<MatchResult['screening_result'], string> = {
  PASS: 'success',
  REVIEW: 'warning',
  REJECT: 'error'
};
